File: apresentacao.js
class Pessoa {
    constructor(nome, idade) {
        this.nome = nome;
        this.idade = idade;
    }

    apresentar() {
        return `Olá, pessoal! O meu nome é ${this.nome} e eu tenho ${this.idade} anos. Estou aprendendo desenvolvimento!`;
    }
}

function validarIdade(idade) {
    return typeof idade === "number" && idade > 0 && idade < 120; // Validação robusta
}

// Uso
const nome = "Letícia Rocha";
const idade = 28;

if (validarIdade(idade)) {
    const pessoa = new Pessoa(nome, idade);
    console.log(pessoa.apresentar());
} else {
    console.log("A idade fornecida é inválida.");
}

File: apresentacao.cs
using System;

class Pessoa {
    public string Nome { get; private set; } // Propriedade automática com setter privado
    public int Idade { get; private set; }   // Propriedade automática com setter privado

    public Pessoa(string nome, int idade) {
        Nome = nome;
        Idade = idade;
    }

    public string Apresentar() {
        return $"Olá, pessoal! O meu nome é {Nome} e eu tenho {Idade} anos. Estou aprendendo desenvolvimento!";
    }
}

class Program {
    static bool ValidarIdade(int idade) {
        return idade > 0 && idade < 120; // Regras básicas para idade válida
    }

    static void Main() {
        const string nome = "Letícia Rocha"; // Nome é constante
        const int idade = 28;

        if (ValidarIdade(idade)) {
            Pessoa pessoa = new Pessoa(nome, idade);
            Console.WriteLine(pessoa.Apresentar());
        } else {
            Console.WriteLine("A idade fornecida é inválida.");
        }
    }
}
